sdk/core/azure_core_test/src/recording.rs

```diff
@@ -333,9 +333,10 @@ impl Recording {
             return value;
         }
 
+        let value = value?;
         let mut variables = self.variables.write().map_err(write_lock_error).ok()?;
-        variables.insert(key.into(), Value::from(value.as_ref(), options));
-        value
+        variables.insert(key.into(), Value::from(Some(&value), options));
+        Some(value)
```

Comment:
It took me a couple times to really notice the effect of the `?` on line 336. Probably more obvious e.g.,


```suggestion
        match value {
            None => None,
            // Treat empty variable as absent.
            Some(v) if v.is_empty() => None,
            Some(v) => {
                let mut variables = self.variables.write().map_err(write_lock_error).ok()?;
                variables.insert(key.into(), Value::from(value.as_ref(), options));
                value
            }
        }
```

Reply:
Clippy is actually why the `let value = value?;` line is there :). I had a more complicated let Some() thingy.

But I like your construct better.